container-search/src/main/java/com/yahoo/prelude/query/parser/SpecialTokens.java

```diff
@@ -142,16 +142,23 @@ public String replace() {
             return replace;
         }
 
+        @Override
         public int compareTo(SpecialToken other) {
-            if (this.token().length() < other.token().length()) {
-                return 1;
-            } else if (this.token().length() == other.token().length()) {
-                return 0;
-            } else {
-                return -1;
-            }
+            if (this.token().length() < other.token().length()) return 1;
```

Comment:
longer token sorts before shorter ?
Reverse of normal alphabetical sorting ?

Reply:
Yes, this is odd but I didn't want to change anything here (except in Relevance).

Reply:
Btw, I think the longer tokens first thing is to prefer matching longer tokens.